comit/src/expiries.rs

```diff
@@ -0,0 +1,488 @@
+        }
+
+        let (_action, next_state) = state.next();
+        let transition_period = state.transition_period();
+
+        period_to_complete(config, next_state, acc + transition_period)
+    }
+
+    period_to_complete(config, current_state, Duration::zero())
+}
+
+/// Happy path states for Alice.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AliceState {
+    // FIXME: Do we need a pre-created state in order to give Alice one more TTA period
+    // before `created`?
+    Created,
+    FundAlphaTransactionSent,
+    AlphaFunded, // Implies alpha fund transaction confirmed configured number of times.
+    BetaFunded,  // Implies beta fund transaction confirmed configured number of times.
+    RedeemBetaTransactionSent,
+    Done, // Implies beta redeem transaction confirmed configured number of times.
+}
+
+/// Possible next action for Alice.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AliceAction {
+    // Happy path actions for Alice.
+    FundAlpha,
+    WaitForFundTransactionToBeConfirmed,
+    WaitForBetaFunded,
+    RedeemAlpha,
+    WaitForRedeemAlphaTransactionToBeConfirmed,
+    NoFurtherAction, // Implies swap is done from Alice's perspective.
+    // Cancel path.
+    Abort,         // Implies HTLC not funded, abort but take no further action.
+    WaitForRefund, // Implies alpha ledger HTLC funded but expiry not yet elapsed.
+    Refund,        // Implies HTLC funded and expiry time elapsed.
+}
+
+impl AliceState {
+    /// Gets the next state and the action required to transition to that state.
+    fn next(&self) -> (AliceAction, AliceState) {
+        use self::{AliceAction::*, AliceState::*};
+
+        match self {
+            Created => (FundAlpha, FundAlphaTransactionSent),
+            FundAlphaTransactionSent => (WaitForFundTransactionToBeConfirmed, AlphaFunded),
+            AlphaFunded => (WaitForBetaFunded, BetaFunded),
+            BetaFunded => (RedeemAlpha, RedeemBetaTransactionSent),
+            RedeemBetaTransactionSent => (WaitForRedeemAlphaTransactionToBeConfirmed, Done),
+            Done => (NoFurtherAction, Done),
+        }
+    }
+
+    /// The minimum time we need to allow to transition to the next state.
+    fn transition_period(&self) -> Duration {
+        use self::AliceAction::*;
+
+        let (next_action, _next_state) = self.next();
+        if next_action == NoFurtherAction {
+            return Duration::zero();
+        }
+
+        unimplemented!()
+    }
+
+    /// True if Alice has funded the alpha HTLC.
+    fn includes_fund(&self) -> bool {
+        match self {
+            AliceState::Created => false,
+            // Be conservative because whether or not to try and refund depends on this value.
+            AliceState::FundAlphaTransactionSent => true,
+            AliceState::AlphaFunded => true,
+            AliceState::BetaFunded => true,
+            AliceState::RedeemBetaTransactionSent => true,
+            AliceState::Done => true,
```

Comment:
```suggestion
            AliceState::FundAlphaTransactionSent | AliceState::AlphaFunded | AliceState::BetaFunded | AliceState::RedeemBetaTransactionSent | AliceState::Done=> true,
```

Reply:
lol, so obvious now I see it on the screen :)